common/params_pyx.pyx

```diff
@@ -0,0 +1,157 @@
+  b"CarBatteryCapacity": [TxType.PERSISTENT],
+  b"CarParams": [TxType.CLEAR_ON_MANAGER_START, TxType.CLEAR_ON_PANDA_DISCONNECT],
+  b"CarParamsCache": [TxType.CLEAR_ON_MANAGER_START, TxType.CLEAR_ON_PANDA_DISCONNECT],
+  b"CarVin": [TxType.CLEAR_ON_MANAGER_START, TxType.CLEAR_ON_PANDA_DISCONNECT],
+  b"CommunityFeaturesToggle": [TxType.PERSISTENT],
+  b"CompletedTrainingVersion": [TxType.PERSISTENT],
+  b"DisablePowerDown": [TxType.PERSISTENT],
+  b"DisableUpdates": [TxType.PERSISTENT],
+  b"DoUninstall": [TxType.CLEAR_ON_MANAGER_START],
+  b"DongleId": [TxType.PERSISTENT],
+  b"GitBranch": [TxType.PERSISTENT],
+  b"GitCommit": [TxType.PERSISTENT],
+  b"GitRemote": [TxType.PERSISTENT],
+  b"GithubSshKeys": [TxType.PERSISTENT],
+  b"HasAcceptedTerms": [TxType.PERSISTENT],
+  b"HasCompletedSetup": [TxType.PERSISTENT],
+  b"IsDriverViewEnabled": [TxType.CLEAR_ON_MANAGER_START],
+  b"IsLdwEnabled": [TxType.PERSISTENT],
+  b"IsMetric": [TxType.PERSISTENT],
+  b"IsOffroad": [TxType.CLEAR_ON_MANAGER_START],
+  b"IsRHD": [TxType.PERSISTENT],
+  b"IsTakingSnapshot": [TxType.CLEAR_ON_MANAGER_START],
+  b"IsUpdateAvailable": [TxType.CLEAR_ON_MANAGER_START],
+  b"IsUploadRawEnabled": [TxType.PERSISTENT],
+  b"LastAthenaPingTime": [TxType.PERSISTENT],
+  b"LastUpdateTime": [TxType.PERSISTENT],
+  b"LastUpdateException": [TxType.PERSISTENT],
+  b"LiveParameters": [TxType.PERSISTENT],
+  b"OpenpilotEnabledToggle": [TxType.PERSISTENT],
+  b"LaneChangeEnabled": [TxType.PERSISTENT],
+  b"PandaFirmware": [TxType.CLEAR_ON_MANAGER_START, TxType.CLEAR_ON_PANDA_DISCONNECT],
+  b"PandaFirmwareHex": [TxType.CLEAR_ON_MANAGER_START, TxType.CLEAR_ON_PANDA_DISCONNECT],
+  b"PandaDongleId": [TxType.CLEAR_ON_MANAGER_START, TxType.CLEAR_ON_PANDA_DISCONNECT],
+  b"Passive": [TxType.PERSISTENT],
+  b"RecordFront": [TxType.PERSISTENT],
+  b"ReleaseNotes": [TxType.PERSISTENT],
+  b"ShouldDoUpdate": [TxType.CLEAR_ON_MANAGER_START],
+  b"SubscriberInfo": [TxType.PERSISTENT],
+  b"TermsVersion": [TxType.PERSISTENT],
+  b"TrainingVersion": [TxType.PERSISTENT],
+  b"UpdateAvailable": [TxType.CLEAR_ON_MANAGER_START],
+  b"UpdateFailedCount": [TxType.CLEAR_ON_MANAGER_START],
+  b"Version": [TxType.PERSISTENT],
+  b"Offroad_ChargeDisabled": [TxType.CLEAR_ON_MANAGER_START, TxType.CLEAR_ON_PANDA_DISCONNECT],
+  b"Offroad_ConnectivityNeeded": [TxType.CLEAR_ON_MANAGER_START],
+  b"Offroad_ConnectivityNeededPrompt": [TxType.CLEAR_ON_MANAGER_START],
+  b"Offroad_TemperatureTooHigh": [TxType.CLEAR_ON_MANAGER_START],
+  b"Offroad_PandaFirmwareMismatch": [TxType.CLEAR_ON_MANAGER_START, TxType.CLEAR_ON_PANDA_DISCONNECT],
+  b"Offroad_InvalidTime": [TxType.CLEAR_ON_MANAGER_START],
+  b"Offroad_IsTakingSnapshot": [TxType.CLEAR_ON_MANAGER_START],
+  b"Offroad_NeosUpdate": [TxType.CLEAR_ON_MANAGER_START],
+  b"Offroad_UpdateFailed": [TxType.CLEAR_ON_MANAGER_START],
+}
+
+class UnknownKeyName(Exception):
+  pass
+
+cdef class Params:
+  cdef c_Params* p
+
+  def __cinit__(self, d=None):
```

Comment:
might as well add an arg for persistent params to be consistent with the c++ api